components/patient-data/impl/src/test/java/org/phenotips/data/internal/controller/APGARControllerTest.java

```diff
@@ -0,0 +1,259 @@
+
+    private static final String APGAR_1 = "apgar1";
+
+    private static final String APGAR_5 = "apgar5";
+
+    private Logger logger;
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    private APGARController controller;
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private XWikiDocument doc;
+
+    @Mock
+    private BaseObject data;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        MockitoAnnotations.initMocks(this);
+
+        this.controller = (APGARController)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+
+        DocumentReference patientDocument = new DocumentReference("wiki", "patient", "00000001");
+        doReturn(patientDocument).when(this.patient).getDocument();
+        doReturn(this.doc).when(this.documentAccessBridge).getDocument(patientDocument);
+        doReturn(this.data).when(this.doc).getXObject(Patient.CLASS_REFERENCE);
+    }
+
+    @Test
+    public void loadCatchesExceptionWhenPatientDoesNotHavePatientClass()
+    {
+        doReturn(null).when(this.doc).getXObject(Patient.CLASS_REFERENCE);
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        verify(this.logger).error("Could not find requested document or some unforeseen"
+                        + " error has occurred during controller loading ",
+                PatientDataController.ERROR_MESSAGE_NO_PATIENT_CLASS);
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void loadDoesNotReturnNullIntegers()
+    {
+        doReturn(null).when(this.data).getStringValue(anyString());
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void loadDoesNotReturnNonIntegerStrings()
+    {
+        doReturn("STRING").when(this.data).getStringValue(anyString());
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        Assert.assertEquals(0, result.size());
+    }
+
+    @Test
+    public void loadReturnsExpectedIntegers()
+    {
+        doReturn("1").when(this.data).getStringValue(APGAR_1);
+        doReturn("2").when(this.data).getStringValue(APGAR_5);
+
+        PatientData<Integer> result = this.controller.load(this.patient);
+
+        Assert.assertEquals(Integer.valueOf(1), result.get(APGAR_1));
+        Assert.assertEquals(Integer.valueOf(2), result.get(APGAR_5));
+        Assert.assertEquals(2, result.size());
+    }
+
+    @Test
+    public void writeJSONReturnsWhenGetDataReturnsNull() {
+        doReturn(null).when(this.patient).getData(DATA_NAME);
+        JSONObject json = new JSONObject();
+
+        this.controller.writeJSON(this.patient, json);
+
+        Assert.assertTrue(json.getJSONObject(DATA_NAME) == null || json.getJSONObject(DATA_NAME).isNullObject());
```

Comment:
No, one isn't at all (`null`), the other is a special JSONObject that represents `null`.